src/Codeception/Module/Doctrine2.php

```diff
@@ -523,41 +547,44 @@ public function haveInRepository($classNameOrInstance, array $data = [])
     }
 
     /**
-     * @param string $className
+     * @template T of object
+     * @param class-string<T> $className
      * @param array $data
-     * @param array $instances
-     * @return object
+     * @param list<object> $instances
+     * @return T
      * @throws ReflectionException
      */
     private function instantiateAndPopulateEntity(string $className, array $data, array &$instances)
     {
         $rpa = new ReflectionPropertyAccessor();
-        list($scalars,$relations) = $this->splitScalarsAndRelations($className, $data);
+        [$scalars,$relations] = $this->splitScalarsAndRelations($className, $data);
         // Pass relations that are already objects to the constructor, too
         $properties = array_merge(
             $scalars,
             array_filter($relations, function ($relation) {
                 return is_object($relation);
             })
         );
+        /** @var T $instance */
         $instance = $rpa->createWithProperties($className, $properties);
         $this->populateEntity($instance, $data, $instances);
         return $instance;
     }
 
     /**
-     * @param object $instance
+     * @template T of object
+     * @param T $instance
      * @param array $data
-     * @param array &$instances
-     * @return object
+     * @param list<object> $instances
```

Comment:
pass by reference `&` is not necessary in documentation blocks?

Reply:
I'm not sure if it's a correct syntax – `@param array &$instances`... PHPStan doesn't support it: https://phpstan.org/r/690e2d73-4c08-4cd8-8056-6f905951888e

I reported it to PHPStan – https://github.com/phpstan/phpstan/discussions/5366

May we leave it as `@param list<object> $instances`? It doesn't make much sense for me to mark it as a reference in the PHPDoc.

Reply:
> May we leave it as `@param list<object> $instances`?

Ok, thanks.